main.c

```diff
@@ -125,6 +124,44 @@ void updateDisplay(void const *arg) {
 
 void gameLogic(void const *arg) {
 		blinkLEDs = false;
+	// Array of enemies, each with an x and y coordinates, when an enemy is hit it changes the y coordinate to above the screen 
+	// so it can fall again.
+	// number of enemies will always stay the same but their speed of descent will increase the longer the player is alive.
+	// for loop iterates through all the enemies, checks if they are below the bottom (player dies) checks for collision with the bullets
+	// there will only be one to three bullet object. The fire rate will be such that only 1-3 bullets will be on the screen at a time. 
```

Comment:
Instead of a fire rate it might be easier to just keep count of the bullets fired and only allow 3 fired bullets at once and modify the counter based on if the bullet has left the screen / if  the bullet hit an enemy

Reply:
That means the player can just spit out all 3 bullets at almost the same time. Thanks for the suggestion though.